src/lib_json/json_reader.cpp

```diff
@@ -1378,30 +1378,39 @@ bool OurReader::readCppStyleComment() {
   }
   return true;
 }
-
-bool OurReader::readNumber(bool checkInf) {
+// if checkNeg is true, read negative number
+bool OurReader::readNumber(bool checkNeg) {
   const char* p = current_;
-  if (checkInf && p != end_ && *p == 'I') {
-    current_ = ++p;
-    return false;
+  char c = '0'; // stopgap for already consumed character  
+  if(checkNeg){
+    if (checkNeg && p != end_ && *p == 'I') {
```

Comment:
This is for checking against Infinity, right?  Using a flag called "checkNeg" to permit checking against Infinity is confusing. Plus, Infinity is not in the JSON spec, so do we support it but only if its capitalized??

Reply:
This is awkward, it’s repeatedly checked here. And because there's a feature to allow some special floats, so, in some cases, we need to support this.